af_ktls.c

```diff
@@ -286,12 +286,15 @@ static inline struct tls_sock *tls_sk(struct sock *sk)
 	return (struct tls_sock *)sk;
 }
 
-static void increment_seqno(char *s)
+void increment_seqno(unsigned char *seq)
 {
-	u64 *seqno = (u64 *) s;
-	u64 seq_h = be64_to_cpu(*seqno);
-	seq_h++;
-	*seqno = cpu_to_be64(seq_h);
+    int i;
+
+    for (i = 7; i >= 0; i--) {
+        ++seq[i];
+        if (seq[i] != 0)
+            break;
+    }
 }
```

Comment:
That will take some more instructions but looks fine as it is not really a bottleneck. What I noticed though is that this function does not check for wrap-arounds, and it increments the same for TLS and DTLS. In DTLS only the rightmost 48-bits are incremented, the rest is the epoch.